src/pdf.rs

```diff
@@ -0,0 +1,673 @@
+                    day_entry.push(formatted_duration);
+
+                    // Add to total duration
+                    total_duration = add_durations(&total_duration, d);
+                }
+            }
+
+            // Add total duration as the last entry
+            if total_duration.microseconds > 0 {
+                let formatted_total = format_duration(total_duration);
+                day_entry.push(format!("Total: {}", formatted_total));
+            }
+
+            // Add the day's entry to the schedule
+            schedule.push(day_entry);
+        } else {
+            eprintln!("Invalid date: {}-{}-{}", year, month, day);
+        }
+    }
+
+    schedule
+}
+
+fn format_duration(interval: PgInterval) -> String {
+    let total_microseconds = interval.microseconds;
+
+    // Convert microseconds to total seconds
+    let total_seconds = total_microseconds / 1_000_000;
+
+    // Calculate hours and minutes
+    let hours = total_seconds / 3600;
+    let minutes = (total_seconds % 3600) / 60;
+
+    // Return formatted string
+    format!("{:02}:{:02}", hours, minutes)
+}
+
+pub async fn get_month_times(given_month: &str, employee_id: i32) -> sqlx::Result<Vec<Vec<String>>> {
+    let database_pool = set_up_database().await;
+    let next_month = add_month(given_month);
+    let year: i32 = given_month[0..4].parse().expect("Invalid year format");
+    let month: u32 = given_month[5..7].parse().expect("Invalid month format");
+
+    // Query for getting the worktime of the emplyee in given month
+    let datetime_start_stc = format!("{}-01T00:00:00Z", given_month);
+    let datetime_start = DateTime::parse_from_rfc3339(&datetime_start_stc).unwrap();
+
+    let datetime_end_stc = format!("{}-01T00:00:00Z", next_month);
+    let datetime_end = DateTime::parse_from_rfc3339(&datetime_end_stc).unwrap();
+
+    let worktimes = worktime::get_timers_in_boundary(
+        employee_id,
+        datetime_start,
+        datetime_end,
+        &database_pool,
+    )
+    .await?;
+
+    // generate schedule
+    let schedule = generate_schedule(worktimes, year, month);
+
+    Ok(schedule)
+}
+
+pub async fn generate_pdf(given_month: &str, employee_id: i32, first_name: &str, last_name: &str, email: &str) {
+    let (doc, page1, layer1) =
+        PdfDocument::new("Zeiterfassungen", Mm(210.0), Mm(297.0), "Layer 1");
+
+    let font_bold = doc
+        .add_external_font(File::open("fonts/ntn-Bold.ttf").unwrap())
+        .unwrap();
+    let font_medium = doc
+        .add_external_font(File::open("fonts/ntn-Medium.ttf").unwrap())
+        .unwrap();
+    let font_light = doc
+        .add_external_font(File::open("fonts/ntn-Light.ttf").unwrap())
+        .unwrap();
+
+    let mut current_layer = doc.get_page(page1).get_layer(layer1);
+
+    let mut month_time_work = 0;
+    let mut month_time_ride = 0;
+
+    // Header Info's
+    let points = vec![
```

Comment:
Here is something off. The square is not filling out the whole page. Look closely to the right side of the rectangle
![image](https://github.com/user-attachments/assets/ecb4cee6-32d3-4453-b253-318503e7d81c)

Reply:
da kann man actually nichts machen.
idk warum das so ist, aber ich geb die pixel halt so an.
ist auch actually nur bei bestimmten zoomstufen so

Reply:
True auf chrome funktioniert es auch ohne fehler